structs_test.go

```diff
@@ -296,6 +296,33 @@ func TestMap_Anonymous(t *testing.T) {
 	}
 }
 
+func TestMap_Flatnested(t *testing.T) {
+	type A struct {
+		Name string
+	}
+	a := A{Name: "example"}
+
+	type B struct {
+		A `structs:",flatnested"`
+		C int
```

Comment:
Please add here a `Name string` field (the same as in `A`) and also tests how it's get overwritten.